www/polj10/sp2/js/pastebin-rest.js

```diff
@@ -0,0 +1,531 @@
+
+    syntaxSelect.selectedIndex = 148;
+    expirationSelect.selectedIndex = 0;
+    visibilitySelect.selectedIndex = 0;
+
+    visibilitySelect.lastElementChild.disabled = true;
+}
+function closeStatusPanel() {
+    const statusBox = document.getElementById("statusBox");
+    statusBox.className = "";
+    statusBox.classList.add("hidden");
+}
+function showStatusPanel(color, message) {
+    const statusBox = document.getElementById("statusBox");
+    const statusLinkWrap = document.getElementById("statusLinkWrap");
+    statusLinkWrap.innerText = message;
+    statusBox.className = "";
+    statusBox.classList.add(color);
+}
+function assignEventListeners() {
+    document.getElementById("loginButton").onclick = login;
+    document.getElementById("logoutButton").onclick = logout;
+    document.getElementById("createNewPasteButton").onclick = createNewPaste;
+    document.getElementById("closeStatusButton").onclick = closeStatusPanel;
+}
+function init() {
+    assignEventListeners();
+
+    const syntaxSelect = document.getElementById("syntaxSelect");
+    for(let i = 0; i < globals.api_paste_code_names.length; i++)
+    {
+        const opt = document.createElement("option");
+        opt.value= i.toString();
+        opt.innerHTML = globals.api_paste_code_names[i];
+        syntaxSelect.appendChild(opt);
+    }
+    syntaxSelect.selectedIndex = 148;
+
+    const expirationSelect = document.getElementById("expirationSelect");
+    for(let i = 0; i < globals.expire_paste_code_names.length; i++)
+    {
+        const opt = document.createElement("option");
+        opt.value= i.toString();
+        opt.innerHTML = globals.expire_paste_code_names[i];
+        expirationSelect.appendChild(opt);
+    }
+    expirationSelect.selectedIndex = 0;
+
+    const visibilitySelect = document.getElementById("visibilitySelect");
+    for(let i = 0; i < globals.visibility_paste_code_names.length; i++)
+    {
+        const opt = document.createElement("option");
+        opt.value= i.toString();
+        opt.innerHTML = globals.visibility_paste_code_names[i];
+        visibilitySelect.appendChild(opt);
+    }
+    visibilitySelect.selectedIndex = 0;
+
+    if(hasUserApiKey()){
+        getUserInfo();
+    } else {
+        showAnonymousUserId();
+    }
+}
+
+//LOGIN
+function login() {
+    const userName = document.getElementById("loginName").value;
+    const password = document.getElementById("loginPassword").value;
+
+    createApiUserKey(userName, password);
+    closeStatusPanel();
+}
+function logout() {
+    const localStorage = window.localStorage;
+    localStorage.removeItem("api_user_key");
+    showAnonymousUserId();
+}
+
+function createPasteElement(paste_key, paste_date, paste_title, paste_expire_date, paste_private, paste_format_long, paste_url) {
+    const pasteElement = document.createElement("li");
+    pasteElement.classList.add("pasteListElement");
+
+    const linkElement = document.createElement("a");
```

Comment:
nechces to prepsat do jQuery? zacina to byt dost necitelne

Reply:
Nechtěl jsem do toho jQuery zabrušovat, protože jsem nechtěl používat "kanón na vrabce". 
Ušetřím 100KB traffic a nějaký "výpočetní výkon" :)
Pokusím se s tou čitelností ještě něco udělat.

Reply:
zadny kanon to podle me neni, spis "maslo", a s tim traffic to neni uplne pravda, vetsinou se to nacte z cache, protoze jQuery se pouziva skoro vsude a je velmi pravepodobne ze klient uz ho ma stazeny. Ale dobre ze takhle uvazujes, byl jsem taky tak naivni